src/app/child-dev-project/children/children.service.ts

```diff
@@ -322,19 +323,40 @@ export class ChildrenService {
    *
    * Warning: Children without any notes will be missing from this map.
    *
-   * @return A map of childIds as key and days since last note as value
+   * @param forLastNDays (Optional) cut-off boundary how many days into the past the analysis will be done.
+   * @param excludeActivityEvents Whether events tracking participation in recurring activities should be considered
+   * @return A map of childIds as key and days since last note as value;
+   *         For performance reasons the days since last note are set to infinity when larger then the forLastNDays parameter
    */
-  public async getDaysSinceLastNoteOfEachChild(): Promise<Map<string, number>> {
+  public async getDaysSinceLastNoteOfEachChild(
+    forLastNDays: number = 30
+  ): Promise<Map<string, number>> {
+    const startDay = moment().subtract(forLastNDays, "days");
+
     const stats = await this.dbIndexing.queryIndexStats(
-      "notes_index/note_date_in_days_for_child"
+      "notes_index/note_child_by_date",
+      {
+        startkey: [startDay.year(), startDay.month(), startDay.date()],
+      }
     );
 
     const results = new Map();
+    const children = await this.getChildren().pipe(take(1)).toPromise();
```

Comment:
A little thought on this construction: https://stackoverflow.com/a/65506511/10713841
Is this the intended behavior or would `toPromise` be sufficient?.

Reply:
I am intentionally using `take(1)` here, hoping that this will skip the additional computation ChildrenService does to add more information. We only need the list of child ids here.

I would agree that this is rather confusing, however. Probably `ChildrenService` shouldn't emit multiple times